MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm

```diff
@@ -524,18 +529,22 @@ - (void)processCrashReports {
 }
 
 - (void)processLogBufferAfterCrash {
-  NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:self.logBufferDir error:NULL];
-  for (NSString *tmp in files) {
-    if ([[tmp pathExtension] isEqualToString:kMSLogBufferFileExtension]) {
-      NSString *filePath = [self.logBufferDir stringByAppendingPathComponent:tmp];
-      NSData *serializedLog = [NSData dataWithContentsOfFile:filePath];
-      if (serializedLog && serializedLog.length && serializedLog.length > 0) {
-        id<MSLog> item = [NSKeyedUnarchiver unarchiveObjectWithData:serializedLog];
-        if (item) {
-          if ([((NSObject *)item) isKindOfClass:[MSAppleErrorLog class]]) {
-            [self.logManager processLog:item withPriority:self.priority];
-          } else {
-            [self.logManager processLog:item withPriority:MSPriorityDefault];
+
+  // Iterate over priorities, check if we have buffered logs for each one.
+  for (NSInteger priority = 0; priority < kMSPriorityCount; priority++) {
+
+    // Get directory for priority, iterate over each file in it with the kMSLogBufferFileExtension and send
+    // the log if a log can be deserialized.
+    NSString *dirForPriority = [self bufferDirectoryForPriority:(MSPriority)priority];
+    NSArray *files = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:dirForPriority error:NULL];
+    for (NSString *tmp in files) {
+      if ([[tmp pathExtension] isEqualToString:kMSLogBufferFileExtension]) {
+        NSString *filePath = [dirForPriority stringByAppendingPathComponent:tmp];
+        NSData *serializedLog = [NSData dataWithContentsOfFile:filePath];
+        if (serializedLog && serializedLog.length && serializedLog.length > 0) {
+          id<MSLog> item = [NSKeyedUnarchiver unarchiveObjectWithData:serializedLog];
+          if (item) {
+            [self.logManager processLog:item withPriority:(MSPriority)priority];
```

Comment:
I have created a new task for this as this applies to the whole SDK, not just this module.